### ISSUE TITLE
import-next/order autofix moves imports above a shebang, producing a syntax error

### ISSUE THREAD

**Issue by ofri-peretz** [ofri-peretz/eslint#942]
## What happens

`eslint --fix` on a `.mjs` file whose first line is `#!/usr/bin/env node` moves an import **above the shebang**, which is a syntax error. The file is silently corrupted by the autofix.

Before:

```js
#!/usr/bin/env node
/** … */
import { readFileSync } from "node:fs";
import { execFileSync } from "node:child_process";
```

After `--fix`:

```js
import { execFileSync } from "node:child_process";
#!/usr/bin/env node
/** … */
import { readFileSync } from "node:fs";
```

```
Parsing error: '#!' can only be used at the start of a file
```

## Where it was hit

Running the plugins over `apps/engage` in ofri-peretz/blog for the first time. Both node scripts with a hashbang were corrupted in the same run:

- `apps/engage/scripts/control-bands.mjs`
- `apps/engage/scripts/smoke.mjs`

Reverted there; the app's lint setup shipped without the autofix pass.

## Why it matters more than a normal false positive

An autofix that emits invalid code is worse than a wrong report. A wrong report is visible and argued with; a wrong FIX lands in the working tree, and `--fix` is exactly what people run without reading the diff. Both files here stopped parsing, so every later rule on them reported nothing — the corruption also silences the linter.

## Expected

The hashbang is not a statement and cannot be reordered. `order` should treat a leading `#!` line as fixed and place imports after it — the same way it already respects the top of file.

## Lock

A fixture with a hashbang plus two out-of-order imports, asserting the fixed output still parses and the shebang is still line 1. The output-parses assertion is the one that matters: a fixer test that only compares strings would have passed a version that emits this.
